google-cloud-clients/google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -62,33 +63,49 @@
   private Long generation;
 
   /**
-   * @param maxChannelReopens max number of times to try re-opening the channel if it closes on us unexpectedly.
+   * @param maxChannelReopens max number of times to try re-opening the channel if it closes on us
+   *    unexpectedly.
+   * @param blobSourceOptions BlobSourceOption.userProject if you want to pay the charges (required
+   *    for requester-pays buckets). Note:
+   *      Buckets that have Requester Pays disabled still accept requests that include a billing
+   *      project, and charges are applied to the billing project supplied in the request.
+   *      Consider any billing implications prior to including a billing project in all of your
+   *      requests.
+   *      Source: https://cloud.google.com/storage/docs/requester-pays
+   * @param userProject: the project you want billed (set this for requester-pays buckets). Leave
+   *      empty otherwise.
    */
   @CheckReturnValue
   @SuppressWarnings("resource")
-  static CloudStorageReadChannel create(Storage gcsStorage, BlobId file, long position, int maxChannelReopens)
+  static CloudStorageReadChannel create(Storage gcsStorage, BlobId file, long position, int maxChannelReopens, String userProject, BlobSourceOption... blobSourceOptions)
```

Comment:
But then we have the trouble of extracting the userProject from the blobSourceOptions in order to pass it to fetchSize so it can translate it to a BlobGetOption. That doesn't seem better to me.

Reply:
Yeah I guess you have a point here. Still I don't feel like this is optimal (it's very likely it is though) mostly because this method may become very long should we needed to add more field in the future. @pongad do you have a better suggestion here?